### PR TITLE
Add APNS client

### PR DESCRIPTION
**Pull request by wangmengyan95**

### CODE REVIEW

**Comment by wangmengyan95** [comment]
cc @fantasist for review

File: APNS.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  this.sender.on("socketError", console.error);
+}
+
+/**
+ * Send apns request.
+ * @param {Object} data The data we need to send, the format is the same with api request body
+ * @param {Array} deviceTokens A array of device tokens
+ * @returns {Object} A promise which is resolved immediately
+ */
+APNS.prototype.send = function(data, deviceTokens) {
```
Comment (grantland) [inline APNS.js]: no max for `deviceTokens`?

Comment (wangmengyan95) [inline APNS.js]: No, internally the library just sends the notifications one by one, so we do not need to have size limitation.

Comment (grantland) [inline APNS.js]: :+1:

**Comment by bnham** [comment]
LGTM. One issue is how to deal with dev vs. prod device tokens. You'll probably need a higher layer that selects the appropriate APNs connection to use (dev or prod) based on the bundle id of the app (available in the "appIdentifier" field of the installation).

**Comment by bnham** [comment]
Okay, seems fine to me. We can add the connection-selecting behavior in some higher-level module.

**Pull request merged**